hack/release-from/action.go

```diff
@@ -0,0 +1,263 @@
+func publishHashreleaseOperator(version string, archs []string) error {
+	multiArchTags := []string{}
+	for _, arch := range archs {
+		tag := fmt.Sprintf("%s/%s:%s-%s", quayRegistry, imageName, version, arch)
+		if _, err := runCommand("docker", []string{"push", tag}, nil); err != nil {
+			return err
+		}
+		logrus.WithField("tag", tag).Debug("Pushed image")
+		multiArchTags = append(multiArchTags, tag)
+	}
+	image := fmt.Sprintf("%s/%s:%s", quayRegistry, imageName, version)
+	cmd := []string{"manifest", "create", image}
+	for _, tag := range multiArchTags {
+		cmd = append(cmd, "--amend", tag)
+	}
+	if _, err := runCommand("docker", cmd, nil); err != nil {
+		return err
+	}
+	if _, err := runCommand("docker", []string{"manifest", "push", "--purge", image}, nil); err != nil {
+		return err
+	}
+	logrus.WithField("image", image).Debug("Pushed manifest")
+
+	initImage := fmt.Sprintf("%s/%s-init:%s", quayRegistry, imageName, version)
+	if _, err := runCommand("docker", []string{"push", initImage}, nil); err != nil {
+		return err
+	}
+	logrus.WithField("image", initImage).Debug("Pushed init image")
+	return nil
+}
+
+// updateConfig updates the version of image(s) in a config file
+func updateConfig(repoRootDir, configFile string, updates []string) error {
+	components := make(map[string]string)
+	for _, override := range updates {
+		parts := strings.Split(override, ":")
+		if len(parts) != 2 {
+			return fmt.Errorf("Invalid override: %s", override)
+		}
+		components[parts[0]] = parts[1]
+	}
+	// open file locally
+	localFile := fmt.Sprintf("%s/%s", repoRootDir, configFile)
+	var config productConfig
+	if data, err := os.ReadFile(localFile); err != nil {
+		return fmt.Errorf("Error reading local file %s: %s", configFile, err)
+	} else if err = yaml.Unmarshal(data, &config); err != nil {
+		return fmt.Errorf("Error unmarshalling local file %s: %s", configFile, err)
+	}
+	for c, ver := range components {
+		if _, ok := config.Components[c]; ok {
+			config.Components[c] = component{
+				Image:    config.Components[c].Image,
+				Version:  ver,
+				Registry: config.Components[c].Registry,
+			}
+		}
+	}
+	// overwrite local file with updated config
+	if err := os.WriteFile(localFile, []byte(fmt.Sprintf("%s\n", config)), 0o644); err != nil {
+		return fmt.Errorf("Error overwriting local file %s: %s", configFile, err)
+	}
+	return nil
+}
+
+// retrieveBaseVersionConfig gets the config based to use
+func retrieveBaseVersionConfig(baseVersion, repoRootDir string) error {
+	url, err := generateBaseConfigDownloadURL(baseVersion)
+	if err != nil {
+		return fmt.Errorf("Error getting download URL: %s", err)
+	}
+
+	for _, file := range []string{calicoConfig, enterpriseConfig} {
+		// open file locally
+		localFile := fmt.Sprintf("%s/%s", repoRootDir, file)
+		out, err := os.OpenFile(localFile, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0o644)
+		if err != nil {
+			return fmt.Errorf("Error opening local file %s: %s", file, err)
+		}
+		defer out.Close()
```

Comment:
Suggestion: use more descriptive variable names so that it's easier to know at a glance what is being passed around. For example:

```suggestion
	for _, configFileName := range []string{calicoConfig, enterpriseConfig} {
		// open file locally
		localConfigFilePath := fmt.Sprintf("%s/%s", repoRootDir, configFileName)
		outputConfigFile, err := os.OpenFile(localConfigFilePath, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0o644)
		if err != nil {
			return fmt.Errorf("Error opening local file %s: %s", configFilePath, err)
		}
		defer outputConfigFile.Close()
```

Clarifying variable names makes it a lot easier to tell at a glance what something is. In this case 'file' is a string, 'localFile' is a string, and 'out' is a file, meaning none of these variable identifiers are correct.

This is true in multiple other functions, but I just happened to call it out here.